internal/charmstore/server.go

```diff
@@ -46,6 +58,11 @@ func NewServer(db *mgo.Database, si *SearchIndex, config ServerParams, versions
 	if len(versions) == 0 {
 		return nil, errgo.Newf("charm store server must serve at least one version of the API")
 	}
+	config.IdentityLocation = strings.Trim(config.IdentityLocation, "/")
+	config.IdentityAPIURL = strings.Trim(config.IdentityAPIURL, "/")
+	if config.IdentityLocation == "" && config.IdentityAPIURL != "" {
+		config.IdentityLocation = config.IdentityAPIURL + "/v1/discharger"
```

Comment:
Will log something.
I'm not sure a warning is appropriate though (+ it's more code to write :]) - the log message should make it clear.